py/htm/advanced/frameworks/location/location_network_creation.py

```diff
@@ -30,6 +30,8 @@
 from htm.advanced.frameworks.location.path_integration_union_narrowing import computeRatModuleParametersFromCellCount
 from htm.advanced.support.logging_decorator import LoggingDecorator
 
+np.set_printoptions(formatter={'float': '{: 0.3f}'.format})
```

Comment:
@fcr firstly i tried modify the fragile parser, by adding more conditions for replacing characters, but then i figured out why it is behaving so weirdly... This statement `np.set_printoptions(formatter={'float': '{: 0.3f}'.format})` will set up proper formatting for numpy native conversion to string. See this [question](https://stackoverflow.com/questions/62831299/python-numpy-to-string-weird-whitespaces) that i have started. With this the example works without any of my further modifications.

@dkeeney thanks for explanation, i have concluded the same, that it has difficult background why it is like this. 
Anyway i am just digging around, just trying to run it on my machine, so no further changes now

Reply:
@Zbysekz,  the numpy print option is not relevant here since it is the C++ side that is generating the strings that have to be parsed in region. If it was the Python side generating the arguments there wouldn't even be a need to marshall and unmarshall the arguments.

Reply:
> @Zbysekz, the numpy print option is not relevant here since it is the C++ side that is generating the strings that have to be parsed in region. If it was the Python side generating the arguments there wouldn't even be a need to marshall and unmarshall the arguments.

Maybe i have confused the problem by mentioning only the parser before...

My problem was with running l2l4l6_experiment.py in thing_classification example. The loading of "Things" is done on the python side and then [here](https://github.com/htm-community/htm.core/blob/907d60a44d3acf3348717b308cb1521e774297e4/py/htm/advanced/frameworks/location/location_network_creation.py#L516) it is passed further, then through some decorators and goes to [pyRegion.py](https://github.com/htm-community/htm.core/blob/907d60a44d3acf3348717b308cb1521e774297e4/bindings/py/packaging/src/htm/bindings/regions/PyRegion.py#L340) where the conversion from numpy array to string happens.
The error was raised when code in [advanced/regions/__init__.py](https://github.com/htm-community/htm.core/blob/master/py/htm/advanced/regions/__init__.py) file was trying to convert it to json

Reply:
This is exactly the problem I was talking about. 
The call to [pyRegion.py](https://github.com/htm-community/htm.core/blob/907d60a44d3acf3348717b308cb1521e774297e4/bindings/py/packaging/src/htm/bindings/regions/PyRegion.py#L340) then goes to [py_Engine.cpp](https://github.com/htm-community/htm.core/blob/907d60a44d3acf3348717b308cb1521e774297e4/bindings/py/cpp_src/bindings/engine/py_Engine.cpp#L302), the C++ arguments are converted to default strings which then eventually get to [advanced/regions/init.py](https://github.com/htm-community/htm.core/blob/master/py/htm/advanced/regions/__init__.py) where they need to be marshalled to Python objects.
The default strings are not consistent across C++ libraries. That is where the problem lies.

Reply:
Yes that is unfortunate... but at least with this PR the numpy conversion would not make a problem anymore